opentelemetry-sdk/src/resource/mod.rs

```diff
@@ -105,8 +134,11 @@ impl Resource {
         for detector in detectors {
             let detected_res = detector.detect(timeout);
             for (key, value) in detected_res.into_iter() {
-                // using insert instead of merge to avoid clone.
-                resource.attrs.insert(key, value);
+                // This call ensures that if the Arc is not uniquely owned,
+                // the data is cloned before modification, preserving safety.
+                // If the Arc is uniquely owned, it simply returns a mutable reference to the data.
```

Comment:
This is a constructor, at what point would the inner not be uniquely owned?
That being said is there a reason we shouldn't put the `make_mut` before the loop?

Reply:
I believe it would be safe to move `make_mut` before loop at this point. The `inner` is ref-incremented only during LogData/SpanData creation.